code/game/machinery/rechargestation.dm

```diff
@@ -192,8 +192,8 @@
 					// Engineering
 					if(istype(O,/obj/item/stack/sheet))
 						var/obj/item/stack/sheet/S = O
-						if(S.amount < 50)
-							S.amount += 1 * coeff
+						if(S.amount < S.max_amount)
+							S.amount += round(min(1 * coeff, S.max_amount))

```

Comment:
The second part of the min here should probably be `(S.max_amount - S.amount)`, so it can't go over the max. It would also reset it to max if it somehow goes over, but that's fine.